includes/class-wc-payments-utils.php

```diff
@@ -591,10 +591,16 @@ public static function get_filtered_error_message( Exception $e ) {
 	 * @return  int
 	 */
 	public static function get_filtered_error_status_code( Exception $e ) : int {
+		$status_code = 400;
 		if ( $e instanceof API_Exception ) {
-			return $e->get_http_code() ?? 400;
+			$status_code = $e->get_http_code() ?? 400;
+			// Sometimes hosting companies hijack this status code and return their own predefined error page.
+			// In this case, we want to return a 400 instead, since it will break the flow of the checkout page.
+			if ( 402 === $status_code ) {
+				$status_code = 400;
+			}
 		}
-		return 400;
+		return $status_code;
```

Comment:
What are your thoughts on the following alternative, to decrease the indentation due to the cyclomatic complexity?

```php
		$status_code = null;
		if ( $e instanceof API_Exception ) {
			$status_code = $e->get_http_code();
		}

		// Sometimes hosting companies hijack the 402 status code to return their own predefined error page.
		// In this case, we want to return a 400 instead, since it will break the flow of the checkout page.
		if ( 402 === $status_code ) {
			$status_code = 400;
		}

		return $status_code ?? 400;
```

Reply:
@frosso's suggestion is easier to read 👍

Reply:
Addressed here: ffb36ecf757971f6b954887b450cd0cf3dd8ff58 . Thank you for the feedback.